sdk/keyvault/azsecrets/client.go

```diff
@@ -0,0 +1,1011 @@
+//go:build go1.16
+// +build go1.16
+
+// Code generated by Microsoft (R) AutoRest Code Generator (autorest: 3.4.3, generator: @autorest/go@4.0.0-preview.27)
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package azsecrets
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets/internal"
+)
+
+// Client is the struct for interacting with a KeyVault Secrets instance
+type Client struct {
+	kvClient *internal.KeyVaultClient
+	vaultUrl string
+}
+
+// ClientOptions are the configurable options on a Client.
+type ClientOptions struct {
+	// HTTPClient sets the transport for making HTTP requests.
+	HTTPClient policy.Transporter
```

Comment:
Do we call this HTTPClient everywhere? Not Transport? @jhendrixMSFT

Reply:
Yes, I believe this came up before.  We can change the name to Transport.